modules/frontend/searchsharding.go

```diff
@@ -167,6 +167,7 @@ func (s searchSharder) RoundTrip(r *http.Request) (*http.Response, error) {
 
 		// if shouldQuit is true, terminate and abandon requests
 		if progress.shouldQuit() {
+			subCancel()
```

Comment:
nice catch. as we find ourselves pushing trace shards higher and higher in larger clusters this is becoming more important.